src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,38 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { TodoContext } from '../../contexts/TodoContext';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { selectedTodo, setSelectedTodo } = useContext(TodoContext);
+
+  const { todo } = selectedTodo;
+
+  useEffect(() => {
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 300);
+
+    getUser(todo.userId).then(setUser);
+  },
```

Comment:
what will happen if the user's download takes longer than 300ms?
you should still show the loader...

so the logic should be like that

```
  useEffect(() => {
    if (todo.id) {
      setUserLoading(true);
    }

    getUser(todo.userId)
      .then(setUser)
      .finally(() => setUserLoading(false));
  }, [todo.userId]);
  ```